src/client/Client.cc

```diff
@@ -1536,20 +1536,17 @@ mds_rank_t Client::choose_target_mds(MetaRequest *req, Inode** phash_diri)
   return mds;
 }
 
-
 void Client::connect_mds_targets(mds_rank_t mds)
 {
   ldout(cct, 10) << __func__ << " for mds." << mds << dendl;
   ceph_assert(mds_sessions.count(mds));
   const MDSMap::mds_info_t& info = mdsmap->get_mds_info(mds);
-  for (set<mds_rank_t>::const_iterator q = info.export_targets.begin();
-       q != info.export_targets.end();
-       ++q) {
-    if (mds_sessions.count(*q) == 0 &&
-	mdsmap->is_clientreplay_or_active_or_stopping(*q)) {
+  for (const auto &rank : info.export_targets) {
+    if (mds_sessions.count(rank) == 0 &&
+	mdsmap->is_clientreplay_or_active_or_stopping(rank)) {
       ldout(cct, 10) << "check_mds_sessions opening mds." << mds
-		     << " export target mds." << *q << dendl;
-      _open_mds_session(*q);
+		     << " export target mds." << rank << dendl;
+      _open_mds_session(rank);
```

Comment:
Is the above change superfluous? I saw that @batrick mentioned that he discouraged converting loops to use iterators like this. Should this delta be dropped or is there something I'm missing in the changes there?

Reply:
(As we talked about offline,) It's the reverse. New code should get rid of iterators and replace with `for (const auto& foo : bar)` constructs.

Reply:
Yeah, this was what @batrick suggested it should to be in all my PRs :-)